pyquil/api/errors.py

```diff
@@ -0,0 +1,137 @@
+
+class ApiError(Exception):
+    def __init__(self, server_status, explanation):
+        Exception.__init__(self, server_status)
+        self.server_status = server_status
+        self.explanation = explanation
+
+    def __repr__(self):
+        return repr(str(self))
+
+    def __str__(self):
+        return self.server_status + "\n" + self.explanation
+
+
+class CancellationError(ApiError):
+    def __init__(self, server_status):
+        explanation = "Please try resubmitting the job again."
+        ApiError.__init__(self, server_status, explanation)
```

Comment:
Why use this direct call over `super(...).__init__`?

Reply:
I couldn't get that to work for some reason... will try again and see

Reply:
Turns out it was an unrelated Jupyter notebook issue.